lib/rubygems/ext/ext_conf_builder.rb

```diff
@@ -50,11 +50,21 @@ def self.build(extension, dest_path, results, args=[], lib_dir=nil, extension_di
   end
 
   def self.get_relative_path(path, base)
-    path.split("/").zip(base.split("/")).inject(String.new) do |result, (path_component, base_component)|
-      next result if path_component == base_component
-      result.prepend("../") if base_component
-      result.concat("#{path_component}/") if path_component
-      result
+    absolute_path = File.expand_path(path)
+    absolute_base = File.expand_path(base)
+
+    backtracks = 0
+    prefix = absolute_base
+    until absolute_path.start_with?(prefix)
```

Comment:
Won't this be incorrect for eg `base=/a/b, path=/a/bc` ?

Reply:
Blah, yeah probably.

"This should be easy enough to fix" he though...

Maybe we should go back to splitting on "/" after all. The approach we had started with was, roughly:

```
path_parts = path.split("/")
base_parts = base.split("/")

while path_parts[0] == base_parts[0]
  path_parts.unshift
  base_parts.unshift
end

backtracks = [".."] * base_parts.size
File.join(backtracks, path_parts)
```

Reply:
I guess that's getting kinda close to the Pathname implementation, so maybe we want something pretty close to that

Reply:
Yeah, sure! I could've also used `pathname` directly, but introducing stdlib dependencies usually gives us trouble, so I avoided doing it just for this.

Reply:
Alright, pushed up something new. 🤞🏻 

Is it fine to split on "/"? That seems like it wouldn't necessarily work on windows, but I'm not too familiar with how that all works. @gabriel-amaral came up with another approach that doesn't involve splitting on "/", but I don't know if it's necessary:

```rb
  def self.get_parts(path)
    parts = []
    prefix = path
    until File.dirname(prefix) == prefix
      dirname, basename = File.split(prefix)
      parts << basename
      prefix = dirname
    end
    parts.reverse
  end
```

Reply:
I guess let's go with the alternative approach just in case. We could split by `File::SEPARATOR`, but there's also `File::ALT_SEPARATOR` so we'd need to check both and it seems it would get messy?

Reply:
We could add a method next to `Gem.path_separator` that's like `Gem.file_separator`, and if ALT_SEPARATOR is defined then cache a `Regexp.union` of the two?